custom/enikshay/nikshay_datamigration/factory.py

```diff
@@ -206,6 +213,17 @@ def test(self, followup):
 
     @property
     @memoized
+    def nikshay_id_to_preexisting_nikshay_person_cases(self):
+        return {
+            person_case.dynamic_case_properties()['nikshay_id']: person_case
+            for person_case in self.case_accessor.get_cases([
+                case_id for case_id in self.case_accessor.get_case_ids_in_domain(type='person')
+            ])
+            if person_case.dynamic_case_properties().get('migration_created_case')
```

Comment:
did we ever talk about using `get_case_by_external_id` for this? would mean sticking `nikshay_id` in the `external_id` field... not sure if we are using that for anything else currently. that would be nice in that it wouldn't have to load every person case in the domain from the DB. we could also consider using ES, though it's not primary data which is always a little scary.